File: bot_saves_princes_2/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

string nextMove(int n, int r, int c, vector <string> grid){
    // Find the princess
    unsigned int row, col;
    for (row = 0;  row < grid.size(); row++) {
        col = grid[row].find('p');
        if (col != string::npos) break;
    }

    // If princess above, move up
    if (r > row) return "UP";
    // If princess bellow, move down
    else if (r < row) return "DOWN"; 
    // If princess to right, move right
    else if (c < col) return "RIGHT";
    // Otherwise, move left
    return "LEFT";
}

int main(void) {

    int n, r, c;
    vector <string> grid;

    cin >> n;
    cin >> r;
    cin >> c;

    for(int i=0; i<n; i++) {
        string s; cin >> s;
        grid.push_back(s);
    }

    cout << nextMove(n, r, c, grid) << std::endl;
    return 0;
}

File: bot_saves_princes_2/solution.py
#

def nextMove(n,r,c,grid):
    p_row = p_col = 0
    for row, line in enumerate(grid):
        p_col = line.find('p')
        if p_col != -1:
            p_row = row
            break

    if r > p_row:
        return 'UP'
    elif r < p_row:
        return 'DOWN'
    elif c < p_col:
        return 'RIGHT'
    else:
        return 'LEFT'    

n = int(input())
r,c = [int(i) for i in input().strip().split()]
grid = []
for i in range(0, n):
    grid.append(input())

print(nextMove(n,r,c,grid))